docs_src/design/adr/security/0008-Secret-Creation-and-Distribution.md

```diff
@@ -0,0 +1,364 @@
+
+   This option is the most widely supported secret distribution mechanism by container orchestrators.
+
+   EdgeX supports runtime environments such as standard Docker and snaps
+   that have no built-in secret management features.
+   
+   * Generic Docker does not have a built-in secrets mechanism.
+     Manual configuration of a _SECRETSLOC_ should utilize either
+     a host file file system path or
+     a Docker volume.
+   
+   * Snaps also do not have a built-in secrets mechanism.
+     The options for _SECRETSLOC_ are limited
+     to designated snap-writable directories.
+   
+   For comparison:
+
+   * Docker Swarm:
+   Swarm swarm mode is not officially supported by the EdgeX project.
+   Docker Swarm secrets are shared via the `/run/secrets` volume,
+   which is a Linux `tmpfs` volume created on the host and shared with the container.
+   For an example of Docker Swarm secrets, see the
+   [docker-compose secrets stanza](https://docs.docker.com/compose/compose-file/#secrets).
+   Secrets distributed in this manner become part of the RaftDB,
+   and thus it becomes necessary to enable swarm autolock mode,
+   which prevents the Raft database encryption key
+   from being stored plaintext on disk.
+   Swarm secrets have an additional limitation in that they are not
+   mutable at runtime.
+
+   * Kubernetes:
+   Kubernetes is not officially supported by the EdgeX project.
+   Kubernetes also supports the secrets volume approach,
+   though the secrets volume can be mounted anywhere in the container namespace.
+   For an example of Kubernetes secrets volumes, see the
+   [Kubernetes secrets documentation](https://kubernetes.io/docs/concepts/configuration/secret/).
+   Secrets distributed in this manner become part of the `etcd` database,
+   and thus it becomes necessary to specify a
+   [KMS provider for data encryption](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/)
+   to prevent `etcd` from storing plaintext versions of secrets.
+
+## Consequences
+
+As the existing implementation is not fully-compliant with this ADR,
+significant scope will be added to current and future EdgeX releases
+in order to bring the project into compliance.
+
+List of needed improvements:
+
+- PKI private keys
+  * All: Move to using Vault as system of origin for the PKI instead of the standalone `security-secrets-setup` utility.
+  * All: Cache the PKI for Consul and Vault on persistent disk; rotate occasionally.
+  * All: Investigate hardware protection of cached Consul and Vault PKI secret keys.  (Vault cannot unseal its own TLS certificate.)
+  * Snaps: Relocate PKI into _SECRETSLOC_ instead of its own folder.
+
+- Special case: Bring-your-own external Kong certificate and key
+  * The Kong external certificate and key is already stored in Vault,
+    however, additional metadata is needed
+    to signal whether these are auto-generated or manually-installed.
+    A manually-suppinstalledlied certificate and key
```

Comment:
manually-suppinstalledlied (sic)

Reply:
Clearly I meant to replace the entire word, not add one word inside of another.  Fixed.